pkg/app/piped/executor/wait/wait.go

```diff
@@ -83,21 +83,24 @@ func (e *Executor) Execute(sig executor.StopSignal) model.StageStatus {
 	defer ticker.Stop()
 
 	e.LogPersister.Infof("Waiting for %v...", duration)
-	select {
-	case <-timer.C:
-		break
-
-	case <-ticker.C:
-		e.LogPersister.Infof("%v elapsed...", time.Since(startTime))
-
-	case s := <-sig.Ch():
-		switch s {
-		case executor.StopSignalCancel:
-			return model.StageStatus_STAGE_CANCELLED
-		case executor.StopSignalTerminate:
-			return originalStatus
-		default:
-			return model.StageStatus_STAGE_FAILURE
+L:
+	for {
+		select {
+		case <-timer.C:
+			break L
+
+		case <-ticker.C:
+			e.LogPersister.Infof("%v elapsed...", time.Since(startTime))
+
+		case s := <-sig.Ch():
+			switch s {
+			case executor.StopSignalCancel:
+				return model.StageStatus_STAGE_CANCELLED
+			case executor.StopSignalTerminate:
+				return originalStatus
+			default:
+				return model.StageStatus_STAGE_FAILURE
+			}
```

Comment:
nit: Could you please change to use a `return` instead of the `break` at line 90.

``` golang
for {
   select {
   case <- timer.C:
      	e.LogPersister.Infof("Waited for %v", totalDuration)
	return model.StageStatus_STAGE_SUCCESS
   
   case <-ticker.C:
        e.LogPersister.Infof("%v elapsed...", time.Since(startTime))

   case s := <-sig.Ch():
        ....
   }
}
}
```

Reply:
thanks, I fixed it in `git commit --amend`